boards/itsybitsy_m4/src/lib.rs

```diff
@@ -1,84 +1,395 @@
+pub type I2C = I2CMaster2<Sda, Scl>;
+
+/// Convenience for setting up the labelled SDA, SCL pins to
+/// operate as an I2C master running at the specified frequency.
+pub fn i2c_master(
     clocks: &mut GenericClockController,
-    bus_speed: F,
-    sercom1: pac::SERCOM1,
+    baud: impl Into<Hertz>,
+    sercom2: pac::SERCOM2,
     mclk: &mut pac::MCLK,
-    port: &mut Port,
-) -> SPIMaster1<
-    hal::sercom::Sercom1Pad3<gpio::Pb23<gpio::PfC>>,
-    hal::sercom::Sercom1Pad0<gpio::Pa0<gpio::PfD>>,
-    hal::sercom::Sercom1Pad1<gpio::Pa1<gpio::PfD>>,
-> {
-    pins.spi_master(clocks, bus_speed, sercom1, mclk, port)
+    sda: impl Into<Sda>,
+    scl: impl Into<Scl>,
+) -> I2C {
+    let gclk0 = clocks.gclk0();
+    let clock = &clocks.sercom2_core(&gclk0).unwrap();
+    let baud = baud.into();
+    let sda = sda.into();
+    let scl = scl.into();
+    I2CMaster2::new(clock, baud, sercom2, mclk, sda, scl)
+}
+
+/// UART Pads for the labelled UART peripheral
+pub type UartPads = uart::Pads<Sercom3, IoSet3, UartRx, UartTx>;
+
+/// UART device for the labelled RX & TX pins
+pub type Uart = uart::Uart<uart::Config<UartPads>, uart::Duplex>;
+
+/// Convenience for setting up the labelled RX, TX pins to
+/// operate as a UART device running at the specified baud.
+pub fn uart(
+    clocks: &mut GenericClockController,
+    baud: impl Into<Hertz>,
+    sercom3: pac::SERCOM3,
+    mclk: &mut pac::MCLK,
+    uart_rx: impl Into<UartRx>,
+    uart_tx: impl Into<UartTx>,
+) -> Uart {
+    let gclk0 = clocks.gclk0();
+    let clock = &clocks.sercom3_core(&gclk0).unwrap();
+    let baud = baud.into();
+    let pads = uart::Pads::default().rx(uart_rx.into()).tx(uart_tx.into());
+    uart::Config::new(mclk, sercom3, pads, clock.freq())
+        .baud(baud, BaudMode::Fractional(Oversampling::Bits16))
+        .enable()
+}
+
+/// Convenience for setting up the USB Bus allocator
+#[cfg(feature = "usb")]
+pub fn usb_allocator(
+    usb: pac::USB,
+    clocks: &mut GenericClockController,
+    mclk: &mut pac::MCLK,
+    dm: impl Into<UsbDm>,
+    dp: impl Into<UsbDp>,
+) -> UsbBusAllocator<UsbBus> {
+    use pac::gclk::{genctrl::SRC_A, pchctrl::GEN_A};
+
+    clocks.configure_gclk_divider_and_source(GEN_A::GCLK2, 1, SRC_A::DFLL, false);
+    let usb_gclk = clocks.get_gclk(GEN_A::GCLK2).unwrap();
+    let usb_clock = &clocks.usb(&usb_gclk).unwrap();
+    let (dm, dp) = (dm.into(), dp.into());
+    UsbBusAllocator::new(UsbBus::new(usb_clock, mclk, dm, dp, usb))
 }
 
 /// Convenience for setting up the dotstar LED using bitbang'ed
 /// SPI.
 pub fn dotstar_bitbang<T: CountDown + Periodic>(
-    pins: pins::Dotstar,
-    port: &mut Port,
+    miso: DotStarNC,
+    mosi: DotStarData,
+    clk: DotStarClk,
     timer: T,
-) -> apa102_spi::Apa102<
-    bitbang_hal::spi::SPI<Pa27<Input<PullUp>>, Pb3<Output<PushPull>>, Pb2<Output<PushPull>>, T>,
-> {
-    pins.dotstar(port, timer)
+) -> apa102_spi::Apa102<bitbang_hal::spi::SPI<DotStarNC, DotStarData, DotStarClk, T>> {
```

Comment:
You could just do:
```suggestion
) ->Apa102<bitbang_hal::spi::SPI<DotStarNC, DotStarData, DotStarClk, T>> {
```
since you've already `use`'d it up top. Or, drop the `use` and switch to all `apa102_spi::Apa102` (I personally prefer the latter)